### DISCUSSION TITLE
[Feature] Use different thumbs directory for an external library.

### DISCUSSION THREAD

**Discussion by chodthewacko** [immich-app/immich#8492] [Feature Request]
### The feature

I'm in the process of (re)waiting for thumbnails/face detection to run against a massive external library for a second time,
just so a second user can name different faces.

It has occurred to me that a lot of time (and disk space) could be saved if external libraries used a different thumbs directory, particularly if the thumbnail files are based on disk path/filename.

Because all users share the same thumbnail/face recognition settings, If another user shares the same external library as another user , there should be zero need to regenerate the thumbnails/face detection/etc.   It's all the same info.    

Long term, it would be nice if different users had the option of doing different face recognition settings.  I have problems in particular with 4 of my relatives getting blended, and so I need to crank down my max recognition distance.   Someone else might not care about those 4 particular people and would be happier with a higher recognition distance (which then makes naming people easier for them).     If there are plans to have allow users having different face recognition settings, it would still be nice if the 'general' thumbnails were reused between users, and each user could have different directory for face thumbnails.  

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by jwilleke** [comment]
From what I have read, #5089, the statement
_Because all users share the same thumbnail/face recognition settings_
is not correct.

And your feature seems to conflict with #5089.
Though I think this SHOULD at least be an option.

**Reply by chodthewacko** [reply]
Actually, on further thought, you are partially right.
In Admin->settings, the thumbnails and face recognition settings are for the entire system, not per person.
So All users uploading/importing pictures, at any point in time, are using the same settings.

However, each user stores their own set of thumbnails/faces, and stores their own names for whatever faces they choose.  
If you never change the settings, each user's thumbnails/faces will be (should be) identical to other user's data.  So the base idea is correct.

However, what could happen is you change the system default settings, and then only regenerate the thumbnails/faces for user A.  
User B will then continue to have thumbnails/faces with the "old" settings. 
What gets weird at that point though, is that any new pictures uploaded by either user will use the "new" settings.
Faces detected (or no longer detected!) with the new settings might play strangely with pictures processed with the old settings.

Maybe the best compromise is, in addition to 'Force Re-scan all library files', there's an option of 'copy library data from user X', which works if current user and user X are using the same import path.   It could copy thumbnails, faces detected in those thumbnails which would be identical.   And while you are at it, slap on an option to copy the people too (so you don't have to remerge everything twice!), including names (since people's names generally stay the same).
